apps/dashr-svm/app.R

```diff
@@ -0,0 +1,514 @@
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+source("utils/helperFunctions.R")
+source("utils/reusableComponents.R")
+
+# Read in stored datasets (see data/getData.R to see how data was acquired)
+dataList <- readRDS("data/data.rds")
+
+app <- Dash$new(name = "DashR SVM Explorer")
+
+app$layout(
+  htmlDiv(
+    list(
+      htmlDiv(
+        className = "banner",
+        children = list(
+          htmlDiv(
+            className = "container scalable",
+            children = list(
+              htmlH2(
+                htmlA(
+                  "Support Vector Machine (SVM) Explorer",
+                  href = "https://github.com/plotly/dash-svm",
+                  style = list(
+                    textDecoration = "none",
+                    color = "inherit"
+                  )
+                )
+              ),
+              htmlA(
+                htmlImg(src = "https://user-images.githubusercontent.com/37411533/56820587-55d56e80-681a-11e9-91b6-026e1551b338.png"),
+                href = "https://plot.ly/products/dash/"
+              )
+            )
+          )
+        )
+      ),
+      htmlDiv(
+        id = "body",
+        className = "container scalable",
+        children = list(
+          htmlDiv(
+            id = "app-container",
+            className = "row",
+            children = list(
+              htmlDiv(
+                id = "left-column",
+                className = "three columns",
+                style = list(
+                  minWidth = "24.5%",
+                  maxHeight = "calc(100vh - 85px)",
+                  overflowY = "auto",
+                  overflowX = "hidden"
+                ),
+                children = list(
+                  # card() is defined in utils/reusableComponents.R 
+                  card(
```

Comment:
This is a pretty clever use of user-written functions; I don't think I've seen a Dash app in R with this sort of syntax. I'm curious if this approach could be adapted to emulate something like `sidebarPanel`.

Reply:
Yeah, it's really useful, and it could definitely be used to make more complex components! Can't take the credit for the idea, though, the person who wrote the python app came up with it :)